etcdserver/api/v3rpc/key.go

```diff
@@ -29,25 +32,127 @@ func New(s etcdserver.V3DemoServer) pb.EtcdServer {
 }
 
 func (h *handler) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResponse, error) {
+	if err := checkRangeRequest(r); err != nil {
+		return &pb.RangeResponse{}, err
```

Comment:
```
grpc: Server failed to encode response proto: Marshal called with nil
goroutine 7 [running]:
github.com/golang/glog.stacks(0x647100, 0x0, 0x0, 0x0)
    /Users/xiang/go/src/github.com/golang/glog/glog.go:766 +0xb8
github.com/golang/glog.(*loggingT).output(0x6224a0, 0xc800000003, 0xc8200d0180, 0x601bd2, 0x9, 0x6c, 0x0)
    /Users/xiang/go/src/github.com/golang/glog/glog.go:717 +0x259
github.com/golang/glog.(*loggingT).printf(0x6224a0, 0x3, 0x4a7860, 0x29, 0xc82000bdc0, 0x1, 0x1)
    /Users/xiang/go/src/github.com/golang/glog/glog.go:655 +0x1d4
github.com/golang/glog.Fatalf(0x4a7860, 0x29, 0xc82000bdc0, 0x1, 0x1)
    /Users/xiang/go/src/github.com/golang/glog/glog.go:1145 +0x5d
google.golang.org/grpc/grpclog.(*glogger).Fatalf(0x647de0, 0x4a7860, 0x29, 0xc82000bdc0, 0x1, 0x1)
    /Users/xiang/go/src/google.golang.org/grpc/grpclog/logger.go:108 +0x49
google.golang.org/grpc/grpclog.Fatalf(0x4a7860, 0x29, 0xc82000bdc0, 0x1, 0x1)
    /Users/xiang/go/src/google.golang.org/grpc/grpclog/logger.go:78 +0x5d
google.golang.org/grpc.(*Server).sendResponse(0xc8200129b0, 0xd00100, 0xc820104180, 0xc8200d00c0, 0x3d65a0, 0x0, 0xc82000bd00, 0xc82000bdb0, 0x0, 0x0)
```

This is what I was talking about. I was lazy by trying nil, nil... It turns out that it would not panic if you return a gRPCError.